rasa/importers/importer.py

```diff
@@ -0,0 +1,247 @@
+import os
+from functools import reduce
+from typing import Text, Optional, Union, List, Dict, Any
+import logging
+from rasa import data
+
+from rasa.core.domain import Domain, InvalidDomain
+from rasa.core.interpreter import RegexInterpreter, NaturalLanguageInterpreter
+from rasa.core.training.dsl import StoryFileReader
+from rasa.core.training.structures import StoryGraph
+from rasa.importers import utils
+from rasa.nlu.training_data import TrainingData
+import rasa.utils.io as io_utils
+import rasa.utils.common as common_utils
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingFileImporter:
+    """Common interface for different mechanisms to load training data."""
+
+    async def get_domain(self) -> Domain:
+        """Retrieves the domain which should be used for the training."""
+        raise NotImplementedError()
+
+    async def get_story_data(
+        self,
+        interpreter: "NaturalLanguageInterpreter" = RegexInterpreter(),
+        template_variables: Optional[Dict] = None,
+        use_e2e: bool = False,
+        exclusion_percentage: Optional[int] = None,
+    ) -> StoryGraph:
+        """Retrieves the story data which should be used for the training."""
+
+        raise NotImplementedError()
+
+    async def get_config(self) -> Dict:
+        """Retrieves the configuration which should be used for the training."""
+        raise NotImplementedError()
+
+    async def get_nlu_data(self, language: Optional[Text] = "en") -> TrainingData:
+        """Retrieves the nlu training data which should be used for the training."""
+        raise NotImplementedError()
+
+    @staticmethod
+    def load_from_config(
+        config_path: Text,
+        domain_path: Optional[Text] = None,
+        training_data_paths: Optional[List[Text]] = None,
+        load_only_nlu_data: bool = False,
```

Comment:
I had a lot of warning logs when doing `train nlu`. I'm not having anything similar for `train core`, so I think we don't need that atm.

Reply:
But it might help to speed up training if you just run `rasa train core`, wouldn't it? We would not need to load the NLU data. But, as no NLU data are provided (or they are filtered out), it would not change anything. So, I guess, you are right, we don't need it 😄

Reply:
mhm, the nlu data would still be loaded for the fingerprint. So maybe makes sense to add a core only importer 👍